lib/pure/pegs.nim

```diff
@@ -326,7 +326,7 @@ proc backrefIgnoreStyle*(index: range[1..MaxSubpatterns]): Peg {.
   ## from 1. Ignores style for matching.
   result = Peg(kind: pkBackRefIgnoreStyle, index: index-1)
 
-proc spaceCost(n: Peg): int =
+proc spaceCost(n: Peg): int {.noSideEffect.} =
```

Comment:
```suggestion
func spaceCost(n: Peg): int =
```

Reply:
pegs.nim has many noSideEffect pragma usages, so i used it too. Later the whole file can be updated to use funcs instead.

Reply:
The preferred way is to use `func`, but that can also be done in a later PR.

Reply:
Done.